### PR TITLE
Fix ignored ocamlformat attribute

### LINKED ISSUE #610: Ignored ocamlformat attribute
Formatting
```
[@@@ocamlformat "wrap-comments=true"]

type t =
  | Aaaaaaaaaa (* Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. *)
  | Bbbbbbbbbb
```
ignores the attribute while passing `--wrap-comments` does not.

### PR DESCRIPTION
**Pull request by gpetiot**
Fix #610 
No diff with test_branch.

### CODE REVIEW

**Comment by gpetiot** [comment]
Hmm I don't know why the CI fails (make tests is alright locally, I already checked the max-iters).

**Comment by gpetiot** [comment]
> The nature of the bug this is fixing makes me think that the redundancy between `Cmts.t` and `Fmt_ast.c` is a problem. Would it be a better fix to remove the `conf` field from `Cmts.t` and pass the `Conf.t` to the `fmt_*` functions as necessary?

Yes it would be better, I thought having this duplication was a design choice.

**Comment by jberdine** [comment]
> Yes it would be better, I thought having this duplication was a design 
> choice.

Not to my knowledge. Things churned a lot when the conf had to be 
plumbed throughout the formatting code in order to handle attributes, I 
imagine that this slipped through in the confusion.

**Comment by gpetiot** [comment]
> The nature of the bug this is fixing makes me think that the redundancy between `Cmts.t` and `Fmt_ast.c` is a problem. Would it be a better fix to remove the `conf` field from `Cmts.t` and pass the `Conf.t` to the `fmt_*` functions as necessary?

Fixed in my last commits.

**Comment by Julow** [comment]
In `Fmt_ast`, these functions are called a hundreds of times. Maybe you should create wrapper functions that only take `c` and `loc` as argument to reduce the boilerplate ? Same for fmt_after and before.

File: src/Cmts.ml [hunk 1]
```diff
@@ -479,9 +477,9 @@ let split_asterisk_prefixed (txt, {Location.loc_start}) =
   in
   split_asterisk_prefixed_ 0
 
-let fmt_cmt t cmt =
+let fmt_cmt conf cmt =
   let open Fmt in
-  if not t.conf.wrap_comments then wrap "(*" "*)" (str (fst cmt))
+  if not conf.Conf.wrap_comments then wrap "(*" "*)" (str (fst cmt))
```
[the remark this answers is not in the document]
Comment (gpetiot) [reply src/Cmts.ml]: done.

File: src/Cmts.ml [hunk 2]
```diff
@@ -497,7 +495,7 @@ let fmt_cmt t cmt =
                 | _, Some _ -> str line $ fmt "@,*" ) )
 
 (** Find, remove, and format comments for loc. *)
-let fmt_cmts t ?pro ?epi ?(eol = Fmt.fmt "@\n") ?(adj = eol) tbl loc =
+let fmt_cmts t conf ?pro ?epi ?(eol = Fmt.fmt "@\n") ?(adj = eol) tbl loc =
```
[the remark this answers is not in the document]
Comment (gpetiot) [reply src/Cmts.ml]: done.

File: src/Fmt_ast.ml [hunk 3]
```diff
@@ -2792,18 +2794,18 @@ and fmt_constructor_declaration c ctx ~first ~last:_ cstr_decl =
...
       ( hovbox 2
           ( hvbox 2
-              ( Cmts.fmt c.cmts loc
+              ( Cmts.fmt c loc
                   (wrap_if (is_symbol_id txt) "( " " )" (str txt))
               $ fmt_constructor_arguments_result c ctx pcd_args pcd_res )
           $ fmt_attributes c ~pre:(fmt "@;") ~key:"@" atrs
           $ fmt_docstring_padded c doc )
-      $ Cmts.fmt_after c.cmts pcd_loc )
+      $ Cmts.fmt_after c ~pro:(fmt_or c.conf.wrap_comments "@ " " ") pcd_loc
```
Comment (hhugo) [inline src/Fmt_ast.ml]: given the size of the PR, should this be moved to a different PR ?

Comment (gpetiot) [reply src/Fmt_ast.ml]: I agree, done.
